File: mid/dls.py
import sys

def dls(adj, src, limit, path, vis):
    if limit < 0:
        return
    vis[src] = 1
    path.append(src)
    for v in adj[src]:
        if not vis[v]:
            dls(adj, v, limit - 1, path, vis)

def main():
    print("a b c d e f g")
    print("0 1 2 3 4 5 6")
    
    try:
        input_data = sys.stdin.read().split()
        if not input_data:
            return
            
        it = iter(input_data)
        n = int(next(it))
        m = int(next(it))
        
        adj = [[] for _ in range(n)]
        for _ in range(m):
            x = int(next(it))
            y = int(next(it))
            adj[x].append(y)
            adj[y].append(x)
            
        src = int(next(it))
        limit = int(next(it))
        
        vis = [0] * n
        path = []
        dls(adj, src, limit, path, vis)
        
        print(*(path))
        
        # This mapping is done just to view the output easily.
        mapping = ["a", "b", "c", "d", "e", "f", "g"]
        print(*(mapping[p] for p in path))
        
    except (EOFError, StopIteration, ValueError):
        pass

if __name__ == "__main__":
    main()

File: mid/dls.cpp
#include<bits/stdc++.h>
using namespace std;

void dls(vector<vector<int>>& adj, int src, int limit, vector<int>& path, vector<int>& vis)
{
    if(limit < 0)
        return;
    vis[src] = 1;
    path.push_back(src);
    for(int v : adj[src])
    {
        if(!vis[v])
            dls(adj, v, limit-1, path, vis);
    }
       
}

int main()
{
    cout << "a b c d e f g\n";
    cout << "0 1 2 3 4 5 6\n";
    int n, m;
    cin >> n >> m;
    vector<vector<int>>adj(n);
    int x,y;
    for(int i= 0;i<m;i++)
    {
        cin >> x >> y;
        adj[x].push_back(y);
        adj[y].push_back(x);
    }
    int src, limit; cin >> src >> limit;
    vector<int> vis(n, 0);
    vector<int> path;
    dls(adj, src, limit, path, vis);
    
    for(int i= 0;i<path.size();i++)
        cout << path[i] << " ";
    cout << "\n";
    // This mapping is done just to view the output easily.
    string mapping[] = {"a", "b", "c", "d", "e", "f", "g"};
    for(int i= 0;i<path.size();i++)
        cout << mapping[path[i]] << " ";
    cout << "\n";
}